### PR TITLE
BREAKING CHANGE: switching to ESM-by-default & upgrading actions

### PR DESCRIPTION
**Pull request by dlockhart**
This accomplishes two things that both needed to happen together:

### 1) New Visual Diff dependencies

The main motivation for this change. By switching to the new `main` branch of the actions repo, you pick up the latest visual-diff, mocha@9 and puppeteer@12 -- this will likely case minor visual differences.

### 2) Make this project use ES Modules by default

Mocha 9 (required for the new visual-diff action) requires it, and web projects shouldn't be CommonJS anyway. I've converted the things using CommonJS things (like `require()` and `__dirname`) to their ESM equivalents.

### Other minor updates

While I was in here, I updated a few things like the latest dev-server and test-runner and linting configs.

### Master -> main & 1.0 release

While I was in here I renamed the default branch to `main`. I'm also going to make this a major release which will bump the version to `1.0` and hopefully alleviate the syncing version problems we have in BSI.

### CODE REVIEW

File: user-profile-html-block.js [hunk 1]
```diff
@@ -4,6 +4,14 @@ const userProfileContextAttribute = 'data-userprofile-context';
 
 export class HtmlBlockUserProfileRenderer {
 
+	get canRenderInline() {
```
Comment (annabelsegalld2l) [inline user-profile-html-block.js:7]: Nit: Not sure why these were moved above the render function - is it a best practices thing?

**Review by annabelsegalld2l — APPROVED** [review]
LGTM!

File: user-profile-html-block.js (hunk 1 above)
Comment (dlockhart) [reply user-profile-html-block.js:7]: I upgraded the linter and this is one of the rules. Yeah basically getters before methods as a best practice.

File: README.md [hunk 2]
```diff
@@ -117,27 +117,32 @@ npm run test:headless:watch
...
+npx mocha './test/**/*.visual-diff.js' -t 10000 --golden
 ```
 
 ## Versioning & Releasing
 
-> TL;DR: Commits prefixed with `fix:` and `feat:` will trigger patch and minor releases when merged to `master`. Read on for more details...
+> TL;DR: Commits prefixed with `fix:` and `feat:` will trigger patch and minor releases when merged to `main`. Read on for more details...
 
-The [sematic-release GitHub Action](https://github.com/BrightspaceUI/actions/tree/master/semantic-release) is called from the `release.yml` GitHub Action workflow to handle version changes and releasing.
+The [sematic-release GitHub Action](https://github.com/BrightspaceUI/actions/tree/main/semantic-release) is called from the `release.yml` GitHub Action workflow to handle version changes and releasing.
```
Comment (svanherk) [inline README.md:145]: @dlockhart Heads up, maybe add the `sematic -> semantic` fix to your template while we're touching a ton of repos anyways 😛

**Pull request merged**